lab4/app/src/main/java/com/app/lab4/LongThread.java

```diff
@@ -0,0 +1,55 @@
+package com.app.lab4;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
+
+import java.io.InputStream;
+
+public class LongThread implements Runnable {
```

Comment:
👍  that you separated into a different class and you don't have everything in MainActivity.
nit: a more suggestive name would be better...ImageDownloadRunnable